test/integration/change_edition_type_test.rb

```diff
@@ -1,261 +1,93 @@
+  edition_types = [:transaction_edition, :place_edition, :programme_edition, :simple_smart_answer_edition, :business_support_edition, :answer_edition, :guide_edition, :licence_edition, :video_edition, :completed_transaction_edition, :help_page_edition, :campaign_edition]
 
-  test "should be able to convert an AnswerEdition into a TransactionEdition" do
-    answer = FactoryGirl.create(:answer_edition, state: 'published')
-    visit_edition answer
+  sample_parts = Set.new([{"title" => "PART !", "body" => "This is some edition version text.", "slug" => "part-one"}, {"title" => "PART !!", "body" => "This is some more edition version text.", "slug" =>  "part-two"}])
 
-    within "div.tabbable" do
-      click_on "Admin"
-    end
+  conversions = edition_types.permutation(2).reject { |pair| pair[0] == pair[1] }
 
-    assert page.has_button?("Create as new Transaction edition")
+  conversions.each do |pair|
 
-    click_on "Create as new Transaction edition"
+    from, to = pair
+    should "be able to convert #{from} into #{to}" do
+      edition = FactoryGirl.create(from, state: 'published')
+      sample_parts.each {|part| edition.parts.create(part)} if edition.respond_to?(:parts)
 
-    assert page.has_content?(answer.title)
-    assert page.has_content?("New edition created")
-    assert page.has_content?("Introductory paragraph")
-    assert page.has_content?("Will continue on")
-    assert page.has_content?("Link to start of transaction")
-    assert page.has_content?("More information")
+      visit_edition edition
 
-    within :css, '#edition_more_information_input' do
-      assert page.has_xpath?("//textarea[contains(text(), '#{answer.whole_body}')]"), "Expected to see: #{answer.whole_body}"
-    end
-  end
+      within "div.tabbable" do
+        click_on "Admin"
+      end
 
-  test "should be able to convert an AnswerEdition into a SimpleSmartAnswerEdition" do
-    answer = FactoryGirl.create(:answer_edition, state: 'published', body: "abcde")
-    visit_edition answer
+      select(to.to_s.gsub("_", " ").titleize.gsub(/Edition.*/, 'Edition'), from: 'to')
 
-    within "div.tabbable" do
-      click_on "Admin"
-    end
+      click_on "Change format"
 
-    assert page.has_button?("Create as new Simple Smart Answer edition")
+      assert_text edition.title
+      assert_text "New edition created"
 
-    click_on "Create as new Simple Smart Answer edition"
+      edition_whole_body = edition.whole_body.gsub(/\s+/, " ").strip
 
-    assert page.has_content?(answer.title)
-    assert page.has_content?("New edition created")
-    assert page.has_content?("Question 1")
-
-    within :css, '#edition_body_input' do
-      assert page.has_xpath?("//textarea[contains(text(), '#{answer.whole_body}')]"), "Expected to see: #{answer.whole_body}"
+      if edition.respond_to?(:parts)
+        assert(sample_parts.subset?(Set.new(edition.parts.map { |part| part.attributes.slice("title", "body", "slug") })))
```

Comment:
it's a bit hard to figure out what's going on here. I'd suggest either breaking out some intermediate local variables to help get the intent across, or to introduce a custom assertion method that would help get the intent across.